sljit_src/sljitLir.c

```diff
@@ -1380,6 +1380,82 @@ static SLJIT_INLINE CHECK_RETURN_TYPE check_sljit_emit_op1(struct sljit_compiler
 	CHECK_RETURN_OK;
 }
 
+static SLJIT_INLINE CHECK_RETURN_TYPE check_sljit_emit_atomic_load(struct sljit_compiler *compiler, sljit_s32 op,
+	sljit_s32 base_reg, sljit_s32 data_reg, sljit_s32 temp_reg)
+{
+	if (SLJIT_UNLIKELY(compiler->skip_checks)) {
+		compiler->skip_checks = 0;
+		CHECK_RETURN_OK;
+	}
+
+#if (defined SLJIT_ARGUMENT_CHECKS && SLJIT_ARGUMENT_CHECKS)
+	CHECK_ARGUMENT(SLJIT_MOV <= GET_OPCODE(op) && GET_OPCODE(op) <= SLJIT_MOV32 && ((GET_OPCODE(op) % 2) == 1 || GET_OPCODE(op) == SLJIT_MOV));
+
+	/* all args must be registers */
+	CHECK_ARGUMENT(SLJIT_R0 <= base_reg && base_reg <= SLJIT_NUMBER_OF_REGISTERS);
+	CHECK_ARGUMENT(SLJIT_R0 <= data_reg && data_reg <= SLJIT_NUMBER_OF_REGISTERS);
+	CHECK_ARGUMENT(SLJIT_R0 <= temp_reg && temp_reg <= SLJIT_NUMBER_OF_REGISTERS);
+
+	/* Only SLJIT_32, is allowed. */
+	CHECK_ARGUMENT(!(op & (VARIABLE_FLAG_MASK | SLJIT_SET_Z)));
+
+	compiler->last_flags = GET_FLAG_TYPE(op) | (op & (SLJIT_32 | SLJIT_SET_Z));
+#endif
+#if (defined SLJIT_VERBOSE && SLJIT_VERBOSE)
+	if (SLJIT_UNLIKELY(!!compiler->verbose)) {
+		fprintf(compiler->verbose, "  atomic_load %s%s%s%s ", op1_names[GET_OPCODE(op) - SLJIT_OP1_BASE],
+				!(op & SLJIT_32) ? "" : "32", op1_types[GET_OPCODE(op) - SLJIT_OP1_BASE],
+				!(op & SLJIT_SET_Z) ? "" : ".z");
```

Comment:
I think 32 is first, then op1_names, but please check how it looks like.
Alos `atomic_load%s%s%s%s`, no space.